### ISSUE TITLE
Crashing when merging multiple fonts

### ISSUE THREAD

**Issue by ImCytox** [ocornut/imgui#3888]
I am currently trying to add custom transalations. Therefore I need to support chinese glyphes etc. Calling the "mergeGlyphes" func once works fine and the desired font im using automatically uses chinese glyphes.

```
static auto mergeGlyphes = [&](float size)
{
	ImFontConfig cfg;
	cfg.MergeMode = true;
	ImGui::GetIO().Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\simsun.ttc", size, &cfg, ImGui::GetIO().Fonts->GetGlyphRangesChineseFull());
	ImGui::GetIO().Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\malgun.ttc", size, &cfg, ImGui::GetIO().Fonts->GetGlyphRangesKorean());
	ImGui::GetIO().Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\msgothic.ttc", size, &cfg, ImGui::GetIO().Fonts->GetGlyphRangesJapanese());
	ImGui::GetIO().Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\arial.ttf", size, &cfg, ImGui::GetIO().Fonts->GetGlyphRangesCyrillic());
	ImGui::GetIO().Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\leeluisl.ttc", size, &cfg, ImGui::GetIO().Fonts->GetGlyphRangesThai());
};
```

But now I have many different fonts which I use via ImGui::PushFont() . Thats why I am storing them in an array,
But using mergeGlyphes a second time leads to crash on ImGui_ImplDX11_CreateFontsTexture(). It fails to create a texture and then dereferences it.

```
static auto addFonts = [&](std::vector<ImFont*>* fonts, float size) 
{
	fonts->push_back(ImGui::GetIO().Fonts->AddFontFromFileTTF(XOR("c:\\Windows\\Fonts\\Arial.ttf"), size, NULL, ImGui::GetIO().Fonts->GetGlyphRangesDefault())); // which font doesnt matter
	mergeGlyphes(size); // merge these font glyphes into arial
				
	fonts->push_back(ImGui::GetIO().Fonts->AddFontFromMemoryCompressedTTF((void*)CasanovaScotia_compressed_data, CasanovaScotia_compressed_size, g_gui.g_tab_font_load_size, NULL, ImGui::GetIO().Fonts->GetGlyphRangesDefault()));
	mergeGlyphes(size); // merge these font glyphes into CasanovaScotia // without this call no crash
}

std::vector<ImFont*> g_TabFonts;
addFonts(&g_TabFontst, 19.f); 
````

**Comment by ocornut** [comment]
Texture is likely too large, our current font system is very adequate for this situation.
The fonts documentation has various answers for this:
https://github.com/ocornut/imgui/blob/master/docs/FONTS.md#font-loading-instructions

**Comment by ocornut** [comment]
FYI, work on `dynamic_fonts` branch aimed to be merged in 1.92 (#8465) solves this, as glyphs are rasterized and packed on demand when they are used.
